### PR TITLE
Cycle Colors Feature

### PR DESCRIPTION
**Pull request by 63BeetleSmurf**
#### My Problem
I like to have my notes different colors so I can quickly go to the specific one I need with just a quick glance.
Using Random as the default color didn't quite cut it for me as I would quite often end up with 2 or 3 notes in a row with the same color (which were annoying to close as discussed below), and it would take a while to remember what color of note contained what information.
#### My Solution
Adding the option (Preferences > Notes > Cycle colors) to have the notes cycle through the possible colors (starting with the specified default color) not only ensures new notes are a different color, but once you get used to the color sequence it is amazingly easy to just know what information you have put where at a glance.

### CODE REVIEW

File: usr/share/glib-2.0/schemas/org.x.sticky.gschema.xml [hunk 1]
```diff
@@ -49,6 +49,14 @@
       </description>
     </key>
 
+    <key name='cycle-colors' type='b'>
+      <default>false</default>
+      <summary>Cycle colors</summary>
+      <description>
+        If true, each new not will be a differnt color from the last.
```
Comment (m8) [inline usr/share/glib-2.0/schemas/org.x.sticky.gschema.xml:56]: Small typo in here `differnt`

Comment (63BeetleSmurf) [reply usr/share/glib-2.0/schemas/org.x.sticky.gschema.xml]: Fixed. Thanks for reviewing the code and letting me know :).

**Comment by collinss** [comment]
Hi @63BeetleSmurf sorry it took me so long to get to your pull requests!

I've been thinking about this feature (and after talking with the rest of the Mint team, they agree), and I've concluded that with the addition of this feature, the 'random' color option is really just a less effective version of what you're doing here, and therefore is quite pointless. That being the case, I would prefer to just have this replace the random option entirely. 

More specifically, I'd like to make the following changes:

- use the `default-color` key to control the cycling
  - change the 'random' choice in the schema to 'cycle'
  - change 'Random' to 'Cycle Colors' (or similar) in the settings window
- move cycle handling to Note.__init__()
  - remove 'random' code
- keep track of the last used color in a separate 'hidden' key in the schema (ie. not exposed in the settings window)
  - update it when a new note is created
  - for backwards compatibility, implement a startup check that updates the `default-color` key to 'cycle' if the user had previously selected 'random'

**Comment by 63BeetleSmurf** [comment]
Yeah that's a good point. I'll make those changes and let you know when they're done.

**Comment by 63BeetleSmurf** [comment]
I've reimplemented this feature on a fresh base and created a new pull request: https://github.com/linuxmint/sticky/pull/68

**Pull request closed without merging**